core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -100,8 +101,8 @@ object ConsoleConsumer extends Logging {
         consumer.receive()
       } catch {
         case e: Throwable => {
-          error("Error processing message, stopping consumer: ", e)
-          consumer.stop()
+          error("Error processing message, terminating consumer process: ", e)
+          // Consumer will be stopped in shutdown hook
```

Comment:
@guozhangwang Sorry, `KafkaConsumer.close()` doesn't throw `ConsumerWakeupException` since it catches all exceptions, but it logs an error with the exception stack trace. The exception is thrown from the `poll` in auto-commit.
```
[2015-10-13 23:08:33,321] ERROR Auto offset commit failed. (org.apache.kafka.clients.consumer.internals.Coordinator:435)
org.apache.kafka.clients.consumer.ConsumerWakeupException
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.clientPoll(ConsumerNetworkClient.java:278)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.pollUnsentRequests(ConsumerNetworkClient.java:232)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:177)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:172)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:145)
	at org.apache.kafka.clients.consumer.internals.Coordinator.commitOffsetsSync(Coordinator.java:399)
	at org.apache.kafka.clients.consumer.internals.Coordinator.maybeAutoCommitOffsetsSync(Coordinator.java:432)
	at org.apache.kafka.clients.consumer.internals.Coordinator.close(Coordinator.java:238)
	at org.apache.kafka.clients.ClientUtils.closeQuietly(ClientUtils.java:63)
	at org.apache.kafka.clients.consumer.KafkaConsumer.close(KafkaConsumer.java:1164)
	at org.apache.kafka.clients.consumer.KafkaConsumer.close(KafkaConsumer.java:1145)
	at kafka.consumer.NewShinyConsumer.cleanup(BaseConsumer.scala:59)
	at kafka.tools.ConsoleConsumer$.run(ConsoleConsumer.scala:66)
	at kafka.tools.ConsoleConsumer$.main(ConsoleConsumer.scala:42)
	at kafka.tools.ConsoleConsumer.main(ConsoleConsumer.scala)
```
If we change the interface in BaseConsumer, the exception above will get logged as an error whenever the stop() is unnecessarily invoked. I suppose the new `consumer.stop()` could be a `wakeup()` followed by `poll(0)` that catches and ignores wakeup exception followed by a `close()`.